### PR TITLE
Dashboard: Adds custom kiosk mode

### LINKED ISSUE #36202: Custom Kiosk Mode
This PR will add a custom kiosk mode. TBD

**TODO**

- [x] Buttons related to permissions shouldn't be part of the custom settings.avatar
- [x] Compare the current kiosk mode(s) on [grafana play](https://play.grafana.org/) with this branch and make sure the different configurations correlate with how it works on play.*
- [x] Change so we only apply the configuration when you press the kiosk mode button in the dashboard navigation. It should toggle the configuration when pressing it multiple times.**
- [ ] Figure out how we should save the custom configuration values and what profile the user have selected.
- [x] Fix flaky gorilla TimePicker toggle
- [x] Add toggles for the things that is missing and can't be toggled via the config yet. avatar
- [ ] Animate the top navigation menu when you move the mouse.
- [ ] Add toggles for template variable dropdowns
- [x] For Standar and TV hide the checkboxes and show only a description.

**Suggested implementation: add a `enabled: bool;` property to the `GorillaContextType` (should be false by default) and toggle that property when pressing the kiosk mode button in the dashboard navigation. Also add a check if the configuration is enabled or not in the `useGorillaConfiguration` hook. If it is enabled return the configuration. If it isn't return undefined.

Related issues:
#7450
#25894
#26105

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #36202

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/dashboard/containers/DashboardPage.tsx [hunk 1]
```diff
@@ -307,19 +306,16 @@ export class UnthemedDashboardPage extends PureComponent<Props, State> {
...
-              title={dashboard.title}
-              folderTitle={dashboard.meta.folderTitle}
-              isFullscreen={!!viewPanel}
-              onAddPanel={this.onAddPanel}
-              kioskMode={kioskMode}
-              hideTimePicker={dashboard.timepicker.hidden}
-            />
-          </div>
-        )}
+        <div aria-label={selectors.pages.Dashboard.DashNav.nav}>
```
Comment (torkelo) [inline public/app/features/dashboard/containers/DashboardPage.tsx:309]: Is this div needed outside DashNav can't this aria-label be moved inside the DashNav component?

File: public/app/features/dashboard/containers/DashboardPage.tsx [hunk 2]
```diff
@@ -333,10 +329,12 @@ export class UnthemedDashboardPage extends PureComponent<Props, State> {
           >
             <div className={styles.dashboardContent}>
               {initError && <DashboardFailed />}
-              {!editPanel && kioskMode === KioskMode.Off && (
-                <div aria-label={selectors.pages.Dashboard.SubMenu.submenu}>
-                  <SubMenu dashboard={dashboard} annotations={dashboard.annotations.list} links={dashboard.links} />
-                </div>
+              {!editPanel && (
+                <HidableInDisplayProfile pathInProfile="subMenu">
+                  <div aria-label={selectors.pages.Dashboard.SubMenu.submenu}>
```
Comment (torkelo) [inline public/app/features/dashboard/containers/DashboardPage.tsx:334]: wonder when these can also be moved inside Submenu  (not related to this PR but these wrapper divs just look off :)

File: public/app/features/dashboard/displayProfiles/components/DisplayProfileSettings.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    { value: 'dashNav.timePicker', label: 'Time picker' },
+    { value: 'dashNav.title', label: 'Title' },
+    { value: 'dashNav.tvToggle', label: 'Kiosk mode button' },
+    { value: 'dashNav.addPanelToggle', label: 'Add panel button' },
+    { value: 'dashNav.snapshotToggle', label: 'Snapshot Button' },
+    { value: 'dashNav.starToggle', label: 'Star button' },
+    { value: 'dashNav.sharePanelToggle', label: 'Share Dashboard or Panel Button' },
+    { value: 'dashNav.dashboardSettingsToggle', label: 'Settings button' },
+    { value: 'dashNav.saveDashboardToggle', label: 'Save button' },
+  ];
```
Comment (torkelo) [inline public/app/features/dashboard/displayProfiles/components/DisplayProfileSettings.tsx:66]: other than timePicker and title I do not see a use case (demand) for any of these other toggles. 

The thing I have seen requested are
* Hide panel menu 
* Hide top toolbar but still show time picker (So DashNav with only time picker)
* Show variables (Submenu) 

But might have missed some other requests.

File: public/app/features/dashboard/displayProfiles/profiles/playlist.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    dashboardSettingsToggle: DisplayProfileMode.hidden,
+    saveDashboardToggle: DisplayProfileMode.hidden,
+    snapshotToggle: DisplayProfileMode.hidden,
+    starToggle: DisplayProfileMode.hidden,
+    sharePanelToggle: DisplayProfileMode.hidden,
+    customButtons: DisplayProfileMode.hidden,
+  },
+  subMenu: DisplayProfileMode.hidden,
+  sideMenu: DisplayProfileMode.hidden,
+};
```
Comment (torkelo) [inline public/app/features/dashboard/displayProfiles/profiles/playlist.ts:18]: What is the thinking behind this one, you can run a playlist in kiosk mode so wont they conflict?

**Review by torkelo — COMMENTED** [review]
Really like the new approach with DisplayProfileSettings compared to the old Gorilla wrappers. This seems to add less complexity and maintenance cost  

Do this think this looks very promising, my only concern with merging/moving forward is that it's adding maybe too many options without any clear use case :)

File: public/app/features/dashboard/displayProfiles/profiles/index.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import { DashboardDisplayProfile } from '../types';
+import { customProfile } from './custom';
+import { kioskProfile } from './kiosk';
+import { standardProfile } from './standard';
+import { tvProfile } from './tv';
+
+export enum DisplayProfileId {
+  kiosk = 'kiosk',
```
Comment (torkelo) [inline public/app/features/dashboard/displayProfiles/profiles/index.ts:8]: Typescript style guide normally capitalize enum options

File: public/app/features/dashboard/displayProfiles/types.tsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
+export enum DisplayProfileMode {
+  hidden = 'hidden',
```
Comment (torkelo) [inline public/app/features/dashboard/displayProfiles/types.tsx:2]: Typescript style guide normally capitalize enum options

**Comment by XuanmoFeng** [comment]
how to use the custom mode? can i use any docs?

**Comment by shehenshah14** [comment]
how to use the custom kiosk mode? Could you please provide any documentation

**Pull request closed without merging**